src/Scientist/Internals/Experiment.cs

```diff
@@ -68,19 +72,36 @@ public void Try(string name, Func<T> candidate)
         {
             if (_candidates.ContainsKey(name))
             {
-                throw new InvalidOperationException($"You already have a candidate named {name}. Provide a different name for this test.");
+                throw new InvalidOperationException(
+                    $"You already have a candidate named {name}. Provide a different name for this test.");
             }
             _candidates.Add(name, () => Task.FromResult(candidate()));
         }
 
-        public void Ignore(Func<T, T, bool> block) => 
+        public void Ignore(Func<T, T, bool> block) =>
             _ignores.Add((con, can) => Task.FromResult(block(con, can)));
 
-        public void Ignore(Func<T, T, Task<bool>> block) => 
+        public void Ignore(Func<T, T, Task<bool>> block) =>
             _ignores.Add(block);
 
+        public void AddContext(string key, dynamic data)
+        {
+            _contexts.Add(key, data);
+        }
+
+        public bool TryAddContext(string key, dynamic data)
```

Comment:
Hmm, good point. The idea here is you wouldn't pass  an experiment around and keep adding context in other  places, thus adding a duplicate context is a developer error and thus should throw?

If so, I agree. :smile: In that case, forget what I said. Sorry for the churn here.